include/deal.II/particles/generators.h

```diff
@@ -162,6 +164,84 @@ namespace Particles
         StaticMappingQ1<dim, spacedim>::mapping,
       const unsigned int random_number_seed = 5432);
 
+
+    /**
+     * A function that generates particles at the location of the support points
+     * of a DoFHandler, possibly based on a different Triangulation with respect
+     * to the one used to construct the ParticleHandler.
+     * The total number of particles that is added to the @p particle_handler object is
+     * the number of dofs of the DoFHandler that is passed that are within the
+     * triangulation and whose components are within the ComponentMask.
+     *
+     * @param[in] a DOF handler that may live on another triangulation that is
+     * used to establsh the positions of the particles
+     *
+     * @param[in] A vector that contains all the bounding boxes for all
+     * processor. This vector can be established by first using
+     * 'GridTools::compute_mesh_predicate_bounding_box()' and gathering all the
+     * bounding boxes using 'Utilities::MPI::all_gather.
+     *
+     * @param[in,out] particle_handler The particle handler that will take
+     * ownership of the generated particles.
+     *
+     * @param[in] mapping An optional mapping object that is used to map
+     * the DOF locations. If no mapping is provided a MappingQ1 is assumed.
+     *
+     * @param[in] Component mask of the dof_handler from which this originates
```

Comment:
```suggestion
     * @param[in] components Component mask of the dof_handler from which this originates
```

Reply:
But I'm also not quite sure what the text actually means. What is "this originates"? I think you want to say that you're taking a subset of the support points, namely the ones that correspond to shape functions matching the selected components.